CIP-0118/README-implicit.md

```diff
@@ -0,0 +1,895 @@
+We discuss additional possible solutions to problems outlined
+in CPS-15 that could make use of much of the same validation zones infrastructure,
+but with varying kinds of underspecified transactions.
+
+
+## Specification
+
+
+### Validation zones
+
+A validation zone is a list of transactions. This structure replaces singleton transactions
+as a unit of validation. Specifically, instead of singleton transactions,
+validation zones are
+
+- sent across the Cardano network
+- entered into a mempool, and
+- placed in block bodies.  
+
+When validating a zone (i.e. for placement in the mempool or when validating an
+incoming block), each transaction is checked against the current ledger
+state, however, additional checks
+are done at the zone level.
+
+
+### Ledger Rule Changes
+
+The prototype for the changes can be found
+[here](https://github.com/IntersectMBO/formal-ledger-specifications/compare/polina-babel?expand=1)
+
+#### ZONE
+
+Currently, the top-level transaction-processing transition system in the ledger system is `LEDGERS`.
+It considers a list of transactions in the block body valid whenever `LEDGER` is valid for
+each transaction, checked in the order they appear in that list. We propose replacing `LEDGERS`
+with
+
+`ZONES ⊆ LEnv x LState x List (List Tx) x LState`
+
+as the top level transition, which processes each zone in the block by
+validating it with the transition
+
+`ZONE ⊆ LEnv x LState x List Tx x LState`
+
+The `ZONE` system includes two rules : `ZONE-V` may be applicable when all transactions are tagged with
+`isValid` tag that is `True`, and `ZONE-N` may be applicable otherwise. The reason
+for differentiating phase-2 valid and invalid _zones_ (and not just at the level
+of individual transactions) is that it may not be possible to apply subsequent
+transactions in a zone once a single transaction is found to be phase-2 invalid.
+Recall that  
+it is not possible to exclude only _some_ transactions from a zone in a general way,
+so, unlike the existing design, the mempool is no able to reject individual zone transactions
+dependent on a preceding transaction being phase-2 valid. We design the `ZONE-N`
+rule to deal with this situation in a special way.
+
+The following checks are done by both rules :
+
+1. the sum total of the sizes of all transactions in the zone is less than the `maxTxSize`
+protocol parameter (see [Network Requirements and Changes](#network-requirements-and-changes)), and
+2. the collateral provided by each transaction `tx` for script execution is sufficient
+to cover **all scripts in all transactions preceding `tx` in the zone, including `tx` itself**
+(see [DDoS](#ddos))
+3. the `LEDGERS` transition is valid for the zone environment, transaction list,
+and initial `LState` of `ZONE` transition
+4. all collateral inputs must be in the initial `LState` UTxO set (see [DDoS](#ddos))
+6. no transaction in a zone spends an output of another transaction in the same zone (see [Flash Loans](#flash-loans))
```

Comment:
This restriction is unnecessary as long as the transactions are applied sequentially. They are applied in the same way as the ledger currently applies transactions, except the zone can try applying unbalanced transactions. If it is impossible to do a flash loan with the current ledger rules, then still using those rules in the zone would mean the flash loan is *still* impossible. There is nothing wrong with `tx5` spending an output generated by `tx4`.